editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -140,43 +144,59 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
         trackBackwardsNavigationByNodeId,
         node,
         trackInputErrors,
+        track,
       }}
     >
       {children}
     </Provider>
   );
 
   async function track(
-    direction: AnalyticsLogDirection,
-    analyticsId: number,
     nodeId: string,
+    direction?: AnalyticsLogDirection,
+    analyticsSessionId?: number,
   ) {
     const nodeToTrack = flow[nodeId];
+    const logDirection = direction || detemineLogDirection();
+    const analyticsSession = analyticsSessionId || analyticsId;
 
-    const metadata = getNodeMetadata(nodeToTrack);
+    if (!nodeToTrack || !logDirection || !analyticsSession) {
+      return;
+    }
+
+    const metadata: NodeMetadata = getNodeMetadata(nodeToTrack);
     const nodeType = nodeToTrack?.type ? TYPES[nodeToTrack.type] : null;
     const nodeTitle = extractNodeTitle(nodeToTrack);
+    const isAutoAnswered = breadcrumbs[nodeId]?.auto || false;
+    metadata["isAutoAnswered"] = isAutoAnswered;
 
-    // On component transition create the new analytics log
     const result = await insertNewAnalyticsLog(
-      direction,
-      analyticsId,
+      logDirection,
+      analyticsSession,
       metadata,
       nodeType,
       nodeTitle,
       nodeId,
     );
 
-    const id = result?.data.insert_analytics_logs_one?.id;
-    const newLogCreatedAt = result?.data.insert_analytics_logs_one?.created_at;
+    const { id, data } = result?.data.insert_analytics_logs_one || {};
+
+    if (!id || !data || !data.created_at) {
+      return;
```

Comment:
nit: is there a case to log an Airbrake error here like `Failed to insert analytics log`, or empty return is intended?

Reply:
I was just leaving it empty to act as a guard clause and reduce the need to nest subsequent code inside an `if` although this could be a handy place to catch errors. 

There isn't really an error handling in the `analyticsProvider` at the moment and as this is already quite a big PR I'd be tempted to leave it for now and maybe revisit error handling in it's own ticket?

Reply:
That's totally fair - happy for analytics error handling to be revisited separately / later more comprehensively.